python/packages/ollama/agent_framework_ollama/_chat_client.py

```diff
@@ -0,0 +1,312 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import json
+from collections.abc import AsyncIterable, Callable, Mapping, MutableMapping, MutableSequence, Sequence
+from itertools import chain
+from typing import Any, ClassVar
+
+from agent_framework import (
+    AIFunction,
+    BaseChatClient,
+    ChatMessage,
+    ChatOptions,
+    ChatResponse,
+    ChatResponseUpdate,
+    Contents,
+    DataContent,
+    FunctionCallContent,
+    FunctionResultContent,
+    Role,
+    TextContent,
+    TextReasoningContent,
+    ToolProtocol,
+    UsageDetails,
+    get_logger,
+    use_chat_middleware,
+    use_function_invocation,
+)
+from agent_framework._pydantic import AFBaseSettings
+from agent_framework.exceptions import ServiceInitializationError, ServiceInvalidRequestError, ServiceResponseException
+from agent_framework.observability import use_observability
+from ollama import AsyncClient
+
+# Rename imported types to avoid naming conflicts with Agent Framework types
+from ollama._types import ChatResponse as OllamaChatResponse
+from ollama._types import Message as OllamaMessage
+from pydantic import ValidationError
+
+
+class OllamaSettings(AFBaseSettings):
+    """Ollama settings."""
+
+    env_prefix: ClassVar[str] = "OLLAMA_"
+
+    host: str | None = None
+    chat_model_id: str | None = None
+
+
+logger = get_logger("agent_framework.ollama")
+
+
+@use_function_invocation
+@use_observability
+@use_chat_middleware
+class OllamaChatClient(BaseChatClient):
+    """Ollama Chat completion class."""
+
+    OTEL_PROVIDER_NAME: ClassVar[str] = "ollama"
+
+    def __init__(
+        self,
+        host: str | None = None,
+        client: AsyncClient | None = None,
+        chat_model_id: str | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize an Ollama Chat client.
+
+        Args:
```

Comment:
together with the previous comment: 

```suggestion
        Keyword Args:
```